python/lsst/daf/butler/core/butlerConfig.py

```diff
@@ -0,0 +1,104 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Configuration classes specific to the Butler
+"""
+
+import os
+
+from .config import Config
+from .utils import doImport
+
+__all__ = ("ButlerConfig",)
+
+
+class ButlerConfig(Config):
+    """Contains the configuration for a `Butler`
+
+    The configuration is read and merged with default configurations for
+    the particular classes. The defaults are read from
+    ``$DAF_BUTLER_DIR/config`` using names specified by the appropriate classes
+    for registry and datastore, and including the standard schema and storage
+    class definitions.
+
+    Parameters
+    ----------
+    other : `str`
+        Path to butler configuration YAML file.
+    """
+
+    def __init__(self, other):
+
+        # Create an empty config for us to populate
+        super().__init__()
+
+        # Find the butler configs
+        self.defaultsDir = None
+        if "DAF_BUTLER_DIR" in os.environ:
+            self.defaultsDir = os.path.join(os.environ["DAF_BUTLER_DIR"], "config")
+
+        # Storage classes
+        storageClasses = Config(os.path.join(self.defaultsDir, "storageClasses.yaml"))
+        self.update(storageClasses)
+
+        # Default schema
+        schema = Config(os.path.join(self.defaultsDir, "registry", "default_schema.yaml"))
+        self.update(schema)
```

Comment:
Should we be using `RegistryConfig` here?  ("No" is a perfectly good answer; I'm just curious why not.)

Reply:
Probably. Before I added `ConfigSubset` there wasn't any difference. Now there is so I'll revisit that.

Reply:
Although, these configs have to fit in the global structure so should not have the `schema` stripped. The advantage of using `SchemaConfig` is you get the validation so I have to see if I can read the subset and bind it to the larger config hierarchy.